### ISSUE TITLE
Aluminum Profile Enclosure

### ISSUE THREAD

**Issue by Otann** [jon-harper/clockmaker-project#1]
Hey Jon!

I've explored a bit how to build things out of t-slot extrusions, inspired by RatRig cases and how plates are mounted on Voron printers. It seems to be relatively easy...

As for sourcing — there are companies that cut extrusions to pieces of the specific size and ship it. And there are usually panels of acrylic glass available for sale. (500x500mm or 500x600mm)

I've creates a parametric Fusion file, which is based around acrylic panels, so users would not have to cut them themselves.

I designed the magnetic mounts, so the panels could be quickly snapped on and off. And also so you can put the enclosure on the printer with a connected OmniBox ;) And they are sealed with the insulation tape.

Let me know what you think, if you have time to look at it.

[Folder on Google Drive with Step file](https://drive.google.com/drive/folders/1AFnqZZfM_8rt54uSt9R0xfQZgK-dZy24?usp=share_link)

<img width="930" alt="Screenshot 2023-01-19 at 15 15 00" src="https://user-images.githubusercontent.com/683786/213441023-a1157dd9-3f2e-40b2-b844-403632e5df8b.png">

<img width="605" alt="Screenshot 2023-01-19 at 15 15 30" src="https://user-images.githubusercontent.com/683786/213441043-190b3280-cfa6-49ab-bd59-035ad7b9f8af.png">

**Comment by jon-harper** [comment]
Oh cool! This is a neat idea--my enclosures in the past used wood for the sides and were too heavy to consider magnets.

There are some issues you'll run into with not having completely fixed panels--having wiring come away with a panel is less than ideal. I see the notch you've cut away in one side, though! Is that as a wiring passthru?

Some tips:

- Heat loss: 
  - Add a floor
  - Plan for what kind and how thick your insulation will be
- Doors:
  - Double doors or single?
  - Embedded windows or solid acrylic?
- Remember that not all panels have to be the same thickness or removable!

Feel free to drop by the OmniBox Discord if you want to chat more. I'm also on the ZeroG Discord, which has a really active community around building enclosures. They may also have some really good feedback (and plenty of examples of existing enclosures to borrow ideas from).

**Comment by Otann** [comment]
Thank you so much for the tips and warm feedback 🙏

Correct, the notch is for cabling. I haven't done anything about the wiring harness, so I had to make at least a couple of panels detachable, so that the whole Omnibox can pass though. And without disassembling whole enclosure.

I don't a have a decent solution on how to seal this notch properly, planning to improvise :)

There is a rubber seal on the bottom of the frame, so hoping that would be enough. I am thinking about sliding in a plywood base, but I assume it would act like a drum and amplify the noise.

Since panels are magnetically detachable, I thought there would be no need for the doors :)

**Comment by jon-harper** [comment]
> I don't a have a decent solution on how to seal this notch properly, planning to improvise :)

TPU is your friend here! I use a piece with teeth like below:

![image](https://user-images.githubusercontent.com/11852387/213453864-2db7e4dc-57db-4245-ac6c-5743d88ef9ec.png)

It's for a round cutout but the idea is pretty easily adaptable. The flaps fill most of the gaps that the wiring doesn't block.

You can also make TPU gaskets. They are not ideal, but sometimes you need to seal a gap that off-the-shelf stuff won't quite fit or work with.

> There is a rubber seal on the bottom of the frame, so hoping that would be enough. I am thinking about sliding in a plywood base, but I assume it would act like a drum and amplify the noise.

I've never tried something like a rubber skirt, but if it's a solid seal, it should work. If you do plywood, you can add PIR foam underneath to add more insulation and absorb some of that noise. My personal take is to always add more insulation, but that's an opinion. 😃 

> Since panels are magnetically detachable, I thought there would be no need for the doors :)

Right on, lol. 🤦‍♂️

Oh, and don't forget filtration! Styrene fumes do more than just give bad headaches. 😬 Check out the Nevermore project if you need a quick solution for that.

**Comment by Otann** [comment]
Right! I can print a seal then for the missing part of the acrylic panel and use a TPU insert, brilliant 👌 Thank you for the tip 🙏 

It's not technically a rubber, but some kind of a foam that is used as an insulator or a sealant, I suppose? Tessa is brand in that area, not sure if it is international.

![tesa-50601-single-sided-foam-tape-3mm-step7of7-ap](https://user-images.githubusercontent.com/683786/213457743-73abf344-419d-47d5-b5de-a06badc08df2.jpeg)

Yes, Nevermore is my plan for solving the filtration. Great project!

**Comment by jon-harper** [comment]
I use similar gap-filling foam, too. I forget what brand…the hardware store kind? 🙂

 

Keep me posted and feel free to drop by the OmniBox Discord to chat, too!

**Comment by Otann** [comment]
Hey Jon!

Just wanted to share some progress with you. Turned out magnets were too weak to hold a panel. And the panels I've ordered were not cut to exact size I had. So I had to improvise with what I've got. Here is the current state:

![IMG_0026](https://user-images.githubusercontent.com/683786/227716142-15b896e1-ad59-46ea-aeb2-2aa01d8589c9.jpeg)

The connection between the OmniBox is quite messy, but it works so far.

![IMG_0028](https://user-images.githubusercontent.com/683786/227716179-2a33c33a-6436-4a36-9e28-11ce843a1712.jpeg)

As you can see, the temperature inside gets up to 43°C (about 109°F) and stays at it, when I print ABS. The bed is set to 100°C (212°F)

![IMG_0029](https://user-images.githubusercontent.com/683786/227716290-3176d27c-c96a-4928-aee6-70c35523cf0e.jpeg)

I've managed to install a spool holder inside the box as well and set a custom filament guide that goes strait to the extruder:

![IMG_0031](https://user-images.githubusercontent.com/683786/227716353-dbb1fb23-d07e-45cf-a7c1-57e59a0f977a.jpeg)

And the close up shot of how the plastic is mounted to the frame:

![IMG_0032](https://user-images.githubusercontent.com/683786/227716372-d041bfc2-73a9-44ef-8a59-716511dbe04e.jpeg)

I've used it to print a Stealthburner for my Ender 3v2 and it turned out okay! So little curling that it is barely noticeable.

![IMG_0015](https://user-images.githubusercontent.com/683786/227716530-eecd2300-4006-4e4e-9e48-8e6160bd84c9.jpeg)

Let me know is you want any of the step files for your project.

**Comment by jon-harper** [comment]
That turned out well! What's your goal after this? Cleaning up the wiring, etc. or do you have more plans for the enclosure?

I do have an idea for you, if you're interested. Polyisocyanurate (PIR) foam is a great, cheap insulator. Cladding the top and one or two of the sides with 25mm foam would help you maintain a higher ambient temperature and heat up faster. You'll lose some of the view, though, so it's a trade-off.

Edit: I just saw the foam gasketing all around the panels. That's a nice touch.

**Comment by Otann** [comment]
p.s. I suppose the distinct visible lines are because of my previously used E3D Hemera was poorly mounted on the gantry frame. The Stealthburner produces surprisingly much better prints!

**Comment by Otann** [comment]
My main goal was to be able to print ABS, which requires an enclosure and higher ambient temperature. I have no nasty smell in my room and the ABS prints just fine. The next goal I am not so sure :) 

Would be nice to have a faster mounting solutions for panels, based on some sort of a clasping mechanism. Unscrewing wing nuts is quite fast, but a little bit annoying.

Also would be nice to figure out better wiring, so the frame could be lifted much more easily.

**Comment by Otann** [comment]
Wait, now I have an idea to add a bottom plate and mount all electronics underneath!
Not sure where the screen would go though.

But that could be an idea for your project too ;)

How is your progress with wiring?

**Comment by jon-harper** [comment]
Oh, I didn't realize you don't have a door. I can see how that gets tedious!

My go-to for enclosures and OmniBoxes is either to the side (as you have it) or on top, with the wires coming down in from above. I use both.

All of my projects now use panel mounts for the OmniBox wiring so that I can disconnect and move the printer/enclosure and OmniBox separately. I'm working on partly panel mounting my enclosure, too, so the wires can disconnect from either end. I'm not sure how comfortable you are with crimping connectors, but that really helped with my wiring mess.

Re: electronics underneath - This is an interesting subject. Underneath is very convenient, but also makes the electronics largely inaccessible. I did this for a project and it turned out well, *but* I had to design it specifically so that there was an access hatch under the printer.

Hope some of that is helpful. :-)
